releasing/plugin-commands-deploy/src/createDeployFiles.ts

```diff
@@ -253,48 +257,46 @@ function convertResolvedDependencies (
   return output
 }
 
+function createDepPath (name: string, spec: string): DepPath {
+  const { id } = dp.parseDepPath(spec)
```

Comment:
There's no avoiding `dp.parseDepPath` as the `spec` sometimes contains peers suffix (without `pkg-name@` prefix).

Reply:
I am confused because spec never contains peers suffix

spec is what is specified in the package.json. It is then also duplicated to the lockfile in order to detect when the lockfile is up to date with the package.json. The "versions" in the lockfile can have peer suffixes. The "versions" are in the dependencies, devDependencies and optionalDependencies fields. You can create a depPath from combining the dependency name and the version (also known as ref in code).

Reply:
I have reset the commits and [correct the names](https://github.com/pnpm/pnpm/pull/8926/commits/aaf8b71bf57b7c4d57244e596b15dca8e5715c2c).